db/log_reader.cc

```diff
@@ -191,6 +191,36 @@ bool Reader::ReadRecord(Slice* record, std::string* scratch,
         }
         break;
 
+      case kBadRecordLen:
```

Comment:
Sounds like the implementation is same for both the cases with minor variation. Can you merge it ?